sdk/state/close.go

```diff
@@ -74,7 +74,13 @@ func (c *Channel) CloseTxs() (declTx *txnbuild.Transaction, closeTx *txnbuild.Tr
 			return nil, nil, fmt.Errorf("finding signatures of confirming signer of close tx for declaration tx for latest close agreement: %w", err)
 		}
 		if signed {
-			declTx, err = declTx.AddSignatureDecorated(s)
+			var closeTxHash [32]byte
+			closeTxHash, err = closeTx.Hash(c.networkPassphrase)
+			if err != nil {
+				return nil, nil, fmt.Errorf("hashing close tx for including payload sig in declaration tx: %w", err)
+			}
+			payloadSig := xdr.NewDecoratedSignatureForPayload(s.Signature, s.Hint, closeTxHash[:])
+			declTx, err = declTx.AddSignatureDecorated(payloadSig)
```

Comment:
These functions adding signatures to things are getting pretty gnarly. I think it is worth us trying a refactor where each signature gets its own field. I think it will simplify some things a lot. I'll probably keep the refactor for another time though.